### ISSUE TITLE
Only allow internal traffic for GDPR

### ISSUE THREAD

**Issue by desertking** [ruffle-rs/ruffle#11933]
Hey,

the only sufficient way to make Ruffle GDPR friendly is to allow only internal traffic to the current host. If hosted on domain.com the .swf file loaded should only allow to make calls to domain.com everything else will be blocked so that no url's from third parties can be loaded. This would be the only way to make ruffle GDPR  friendly for the European market. Would appreciate that feature asap :)

**Comment by Dinnerbone** [comment]
I'm not sure what blocking url access has to do with gdpr, but you set `allowNetworking` to `internal` or `none` to achieve this.

**Comment by n0samu** [comment]
@Dinnerbone Ruffle's implementation of `allowNetworking` doesn't currently restrict access to APIs that allow sending data, such as `Loader`. And CORS only restricts access to the response to a request - it doesn't prevent the request from reaching the remote server. So SWFs could still be sending data out to unknown hosts, which could have potential GDPR implications.

**Comment by Dinnerbone** [comment]
GDPR implications there wouldn't be with ruffle itself, but with the hosts of the service that whatever content is sending your personal information to somehow.

**Comment by n0samu** [comment]
True. But site owners have some responsibility for the content hosted on their site as well. Hence the feature request

**Comment by desertking** [comment]
With the cookie banner you ask people to opt in if they don't the only option would be not to load ruffle at all or the swf and block the whole game. So it would be great to have an internal only flag that limits access to apis. You could get sued for even "sending" the ip overseas without consent.
